SingularityUI/app/application.coffee

```diff
@@ -107,6 +107,8 @@ class Application
             if jqxhr.status is 502
                 Messenger().info
                     message:   "Singularity is deploying, your requests cannot be handled. Things should resolve in a few seconds so just hang tight!"
+            else if jqxhr.status is 401 and config.redirectOnUnauthorizedURL
+                window.location.href = config.redirectOnUnauthorizedURL
```

Comment:
we should have some sort of support for redirecting back to the original page once the user has logged in... something along the lines of `.replace('${URL}', window.location.href)` could work